man/Makefile.am

```
@@ -1,72 +1,93 @@
+man_MANS			= \
+	$(generated_md_man)	\
+	$(generated_in_man)
+
+generated_in_man		= \
+	ifcfg-bond.5		\
 	ifcfg-bonding.5		\
 	ifcfg-bridge.5		\
-	ifcfg-dummy.5		\
-	ifcfg-ipoib.5		\
+	ifcfg-dhcp.5		\
 	ifcfg-infiniband.5	\
+	ifcfg-lo.5		\
 	ifcfg-macvlan.5		\
 	ifcfg-macvtap.5		\
 	ifcfg-ovs-bridge.5	\
 	ifcfg-ppp.5		\
-	ifcfg-team.5		\
 	ifcfg-tunnel.5		\
 	ifcfg-vlan.5		\
 	ifcfg-vxlan.5		\
-	ifcfg-wireless.5	\
-	ifcfg-dhcp.5		\
-	ifcfg-lo.5		\
 	ifcfg.5			\
+	ifdown.8		\
 	ifroute.5		\
 	ifrule.5		\
-	ifsysctl.5		\
-	ifdown.8		\
 	ifstatus.8		\
-	ifup.8
+	ifsysctl.5		\
+	ifup.8			\
+	routes.5		\
+	wicked-config.5		\
+	wicked-ethtool.8	\
+	wicked.7		\
+	wicked.8		\
+	wickedd.8
+
+generated_md_man		= \
+	ifcfg-dummy.5		\
+	ifcfg-ipoib.5		\
+	ifcfg-team.5		\
+	ifcfg-wireless.5	\
+	wicked-firmware.8	\
+	wicked-redfish.8
+
 
+templates_md_src		= $(addprefix src/,$(generated_md_man:=.md))
+templates_md_in			= $(generated_md_man:=.in)
+templates_in			= $(generated_in_man:=.in)
 
-templates			= $(generated:=.in)
+# pandoc would write @VAR@ as [at]VAR[at] to .in:
+#   .md sources are using %VAR% variables, while
+#   .in templates are using common autoconf @VAR@
+substitute_md_vars		= \
+	$(AM_V_GEN)sed		-e "s|[%]\(.*\)[%]|@\\1@|g"
 
 substitute_vars			= \
-	$(AM_V_GEN)sed						\
-	-e "s|[@]wicked_piddir[@]|$(wicked_piddir)|g"		\
-	-e "s|[@]wicked_sbindir[@]|$(wicked_sbindir)|g"		\
-	-e "s|[@]wicked_statedir[@]|$(wicked_statedir)|g"	\
-	-e "s|[@]wicked_storedir[@]|$(wicked_storedir)|g"	\
-	-e "s|[@]wicked_schemadir[@]|$(wicked_schemadir)|g"	\
-	-e "s|[@]wicked_configdir[@]|$(wicked_configdir)|g"	\
-	-e "s|[@]wicked_supplicantdir[@]|$(wicked_supplicantdir)|g"\
-	-e "s|[@]wicked_extensionsdir[@]|$(wicked_extensionsdir)|g"\
+	$(AM_V_GEN)sed		\
+	-e "s|[@]wicked_piddir[@]|$(wicked_piddir)|g"			\
+	-e "s|[@]wicked_sbindir[@]|$(wicked_sbindir)|g"			\
+	-e "s|[@]wicked_statedir[@]|$(wicked_statedir)|g"		\
+	-e "s|[@]wicked_storedir[@]|$(wicked_storedir)|g"		\
+	-e "s|[@]wicked_schemadir[@]|$(wicked_schemadir)|g"		\
+	-e "s|[@]wicked_configdir[@]|$(wicked_configdir)|g"		\
+	-e "s|[@]wicked_supplicantdir[@]|$(wicked_supplicantdir)|g"	\
+	-e "s|[@]wicked_extensionsdir[@]|$(wicked_extensionsdir)|g"	\
 	-e "s|[@]PACKAGE_BUGREPORT[@]|$(PACKAGE_BUGREPORT)|g"
 
-wicked%: wicked%.in $(top_builddir)/config.status
-	@echo "substituting variables in $< > $@"
+%: %.in $(top_builddir)/config.status
+	@if $(AM_V_P); then echo "substituting variables: $< > $@" ; fi
 	@rm -f $@ $@.tmp
 	@$(substitute_vars) < $< > $@.tmp && mv $@.tmp $@
 
-if%: if%.in $(top_builddir)/config.status
-	@echo "substituting variables in $< > $@"
-	@rm -f $@ $@.tmp
-	@$(substitute_vars) < $< > $@.tmp && mv $@.tmp $@
+if pandoc
+%.in: src/%.md $(top_builddir)/config.status
+	@if $(AM_V_P); then echo "generating manual page: $< > $@" ; fi
+	@rm -f $@ $@.tmp $@.tmp
+	@$(AM_V_GEN)$(PANDOC) -s -f markdown -t man $< -o $@.tmp
+	@$(substitute_md_vars) -i "$@.tmp" && mv $@.tmp $@
+endif
```

```suggestion
if pandoc
%.in: src/%.md
	@if $(AM_V_P); then echo "generating manual page: $< > $@" ; fi
	@rm -f $@
	@$(AM_V_GEN)$(PANDOC) -s -f markdown -t man $< -o $@
	@$(substitute_md_vars) -i "$@"

else
%.in: src/%.md
	$(error Missing pandoc to regenerate $@ from $< )
endif
```

I would simply error out here if there is no pandoc. Such developer should simply do not toucht the `*.md` files.
I think then this error from 3585f97e533dcb7ee80d5096b145e1500b9d476e is not needed anymore.

And I removed the dependency to `$(top_builddir)/config.status`, not sure why you had it there?

---

> I would simply error out here if there is no pandoc.
No, the suggestion breaks **any build** -- on systems without pandoc and this are all versions < sle15-sp5 and even more: any RPM build, as wicked.spec does not have a BuildRequire enabled by default.

I do not want to convert man pages from md when we build an RPM using the released tar archive (and probably differently due to different pandoc version, ...) -- the .in files are BUILT_SOURCES included in the dist-tar archive created by "make dist" and we simply fill out the variables only while rpmbuild.

> Such developer should simply do not toucht the *.md files.
With the suggestion you don't need to touch the .md files at all. Just try out: `./autogen.sh --disable-pandoc && make` -> error. Same will happen when pandoc is not available (in the build chroot / container, ...) in the system.

> And I removed the dependency to $(top_builddir)/config.status, not sure why you had it there?
This is IMO needed to automatically rebuild when make decides to make a dependency configure re-run while pandoc is available (and we want to rebuild from md). It's required by "distcheck" as the .in files are built sources.